controllers/update_pods.go

```diff
@@ -164,6 +173,22 @@ func shouldRequeueDueToTerminatingPod(pod *corev1.Pod, cluster *fdbv1beta2.Found
 		!cluster.ProcessGroupIsBeingRemoved(processGroupID)
 }
 
+// Because the internal.fault_tolerance relies in fdbcli status only we need to make sure it is
+// synced with the cluster status to prevent race conditions.
+func shouldRequeueDueToClusterStatusDrift(cluster *fdbv1beta2.FoundationDBCluster, adminClient fdbadminclient.AdminClient, podMap map[string]*corev1.Pod) (bool, error) {
+	status, err := adminClient.GetStatus()
+	if err != nil {
+		return false, err
+	}
+	//Check that all processes in fdbcli status have a matching pod.
+	for _, p := range status.Cluster.Processes {
```

Comment:
Thinking more about this case it might be enough to just check if any Pod has a non zero `deletionTimestamp` and is not marked for removal. That would indicate that a deletion/update is already in progress and we want to skip further steps.

Reply:
So we would be adding a check like `shouldRequeueDueToTerminatingPod` but instead of checking `cluster.ProcessGroupIsBeingRemoved` we need to check if the process group is not marked for removal right?

Reply:
Looking again at `shouldRequeueDueToTerminatingPod` that should actually cover the case that you've describe and the default time we block any new updates is 10min. That change was added a while back: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1070 and should be in all versions of 1.0.0+. One thing that we are not checking is if the new Pods are up and running before moving to the next Pod: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/update_pods.go#L71-L76 but that should be catched in the fault tolerance check (since I would expect that FT is reduced in that case).